apps/cowswap-frontend/src/modules/trade/updaters/RecipientAddressUpdater.tsx

```diff
@@ -10,9 +10,9 @@ export function RecipientAddressUpdater() {
 
   useEffect(() => {
     if (state?.recipientAddress !== recipientAddress) {
-      updateState?.({ recipientAddress })
+      updateState?.({ ...state, recipientAddress })
     }
-  }, [recipientAddress, state?.recipientAddress, updateState])
+  }, [recipientAddress, state, updateState])
```

Comment:
This might cause issues with re-rendering.
For the future, we should use the fn update format to not depend on the `state`